### PR TITLE
Expand generated documentation and README docs

### PR DESCRIPTION
**Pull request by horgh**

### CODE REVIEW

File: lib/minfraud/errors.rb [hunk 1]
```diff
@@ -1,11 +1,22 @@
...
+  # The base error class for Minfraud exceptions.
   class BaseError          < StandardError; end
 
+  # An error indicating the value is not valid for the enumerated type.
   class NotEnumValueError  < BaseError; end
+
+  # An error indicating the field is not recognized.
   class RequestFormatError < BaseError; end
+
+  # An error indicating there is a critical problem with the request.
```
Comment (oschwald) [inline lib/minfraud/errors.rb:13]: Don't most of the errors meet this description? It might help if this was a bit more specific. Is this all transport related errors?

Comment (horgh) [reply lib/minfraud/errors.rb]: Good point! I updated it. The errors that lead to this can be seen here: https://github.com/maxmind/minfraud-api-ruby/blob/master/lib/minfraud/error_handler.rb#L30
